src/Server/src/Core/Data/Templates/ClientTemplateService.cs

```diff
@@ -56,9 +56,10 @@ public class ClientTemplateService : UserTenantService, IClientTemplateService
             client.ClientId = GetClientId(template);
             client.Name = GetName(template, application, environment);
             client.ClientUri = GetUri(template, application, environment)?.ToLower();
-            client.RedirectUris = GetRedirectUris(template, application, client, environment);
-            client.AllowedGrantTypes = GetGrantTypes(template, application);
-            client.AllowedScopes = BuildScopes(application);
+            client.RedirectUris = template.RedirectUris.Union(application.RedirectUris)
+                .Select(uri => RenderTemplate(uri, application, environment, client)).ToList();
+            client.AllowedGrantTypes = template.AllowedGrantTypes.Union(application.AllowedGrantTypes).ToList();
```

Comment:
There we go, now we ahve a List on AllowedGrantTypes instead of a set

Reply:
The union will ensure that is no duplicates, it's the same as:
```csharp
new HashSet(template.AllowedGrantTypes.Concat(application.AllowedGrantTypes));
```